velox/exec/tests/TableScanTest.cpp

```diff
@@ -1506,6 +1506,20 @@ TEST_F(TableScanTest, partitionedTableDoubleKey) {
   testPartitionedTable(filePath->path, DOUBLE(), "3.5");
 }
 
+TEST_F(TableScanTest, partitionedTableDecimalKey) {
+  auto rowType = ROW({"c0", "c1"}, {BIGINT(), DOUBLE()});
+  auto vectors = makeVectors(10, 1'000, rowType);
+  auto filePath = TempFilePath::create();
+  writeToFile(filePath->path, vectors);
+  createDuckDbTable(vectors);
+  testPartitionedTable(filePath->path, DECIMAL(12, 3), "123456789123");
+  testPartitionedTable(filePath->path, DECIMAL(12, 3), "-123456789123");
+  testPartitionedTable(
+      filePath->path, DECIMAL(36, 18), "123456789123456789123456789123456789");
```

Comment:
> 123456789123456789123456789123456789

What does this come from?

> Decimal values must be formatted using unscaled values

Maybe this needs to be clarified a bit with an example.

Reply:
`123456789123456789123456789123456789` was just created for test.  Added an example to the comment.

Reply:
Got it. Thanks.

It feels a bit strange that Spark creates directories using negative and floating point numbers, i.e. path/to/-123 or path/to/-1.23 . Does it really do that?

Reply:
Spark has some test cases for that as [link](https://github.com/apache/spark/blob/master/sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetPartitionDiscoverySuite.scala#L696-L697) shows. I assume it is possible if the Parquet is generated partitioning by a decimal/floating column.